packages/firebase_auth/firebase_auth/android/src/main/java/io/flutter/plugins/firebase/auth/FlutterFirebaseAuthPlugin.java

```diff
@@ -1377,9 +1216,9 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
     return Tasks.call(
         cachedThreadPool,
         () -> {
-          removeEventListeners();
+          // TODO: not sure if this is applicable here.
+          //          removeEventListeners();
```

Comment:
When does `didReinitializeFirebaseCore` get called usually?

Reply:
Ah, when the core is reinitialized (doh!) - in the firebase core plugin. If the user changes the firebase project then we need to clean up the plugins. There is a problem in `cloud_firestore` as well as this PR: The `didReinitializeFirebaseCore` does not clear the event listeners, which may cause errors. The listeners get cleared up when the plugin is destroyed. Hm.. 

Sounds like a central cleanup method in the plugin would be the right way to go here.

Reply:
`didReinitializeFirebaseCore` is also called on Dart Hot Restarts in dev (was mainly added for this purpose) - so it would be best if the event listeners are also cleaned up here otherwise these would linger in native when Dart is no longer aware of them?